### PR TITLE
Fixed connections drop under load

### PR DESCRIPTION
**Pull request by const-t**

### CODE REVIEW

**Review by krizhanovsky — APPROVED** [review]
LGTM for me, but I'm not so confident with the `tfw_h2_sk_prepare_xmit()` part, so a very careful review from @EvgeniiMekhanik is required.

File: fw/sock_clnt.c [hunk 1]
```diff
@@ -325,14 +313,16 @@ tfw_h2_sk_prepare_xmit(struct sock *sk, struct sk_buff *skb, unsigned int mss_no
...
+				r = 0;
+				goto update_limit;
+			}
+			goto ret;
 		}
 
 		truesize += tmp_truesize;
 		tmp_truesize = 0;
 
-		skb_set_tfw_flags(skb, SS_F_HTTT2_FRAME_DATA_DONE);
```
Comment (EvgeniiMekhanik) [inline fw/sock_clnt.c:335]: Can we drop SS_F_HTTT2_FRAME_HEADERS_DONE also?

Comment (const-t) [reply fw/sock_clnt.c:335]: Done

**Review by EvgeniiMekhanik — APPROVED** [review]

**Comment by EvgeniiMekhanik** [comment]
LGTM. Please remove SS_F_HTTT2_FRAME_HEADERS_DONE from kernel code and update patch

**Pull request merged**